python/pyspark/sql/window.py

```diff
@@ -124,16 +124,19 @@ def rangeBetween(start, end):
         values directly.
 
         :param start: boundary start, inclusive.
-                      The frame is unbounded if this is ``Window.unboundedPreceding``, or
+                      The frame is unbounded if this is ``Window.unboundedPreceding``,
+                      ``org.apache.spark.sql.catalyst.expressions.UnboundedPreceding``, or
                       any value less than or equal to max(-sys.maxsize, -9223372036854775808).
         :param end: boundary end, inclusive.
-                    The frame is unbounded if this is ``Window.unboundedFollowing``, or
+                    The frame is unbounded if this is ``Window.unboundedFollowing``,
+                    ``org.apache.spark.sql.catalyst.expressions.UnboundedPFollowing``, or
                     any value greater than or equal to min(sys.maxsize, 9223372036854775807).
         """
-        if start <= Window._PRECEDING_THRESHOLD:
-            start = Window.unboundedPreceding
-        if end >= Window._FOLLOWING_THRESHOLD:
-            end = Window.unboundedFollowing
+        if isinstance(start, int) and isinstance(end, int):
```

Comment:
Sure, we need some tests ..

Reply:
This is different from the WindowSpec interface, we support Long values:
```
def rangeBetween(start: Long, end: Long): WindowSpec
```

Reply:
FYI, Python 3 doesn't have `long` and it was merged to `int`. We should do `long` here and assign `int` to `long` in Python 3.

Reply:
oh, thanks @HyukjinKwon , I'm not familiar with python :)

Reply:
@HyukjinKwon @jiangxb1987 
Thank you very much for your comments. 
It seems to me that int and long are "unified" in python 2. I tried the following:
```
Python 2.7.10 (default, Oct 23 2015, 19:19:21) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> value = 9223372036854775807
>>> isinstance(value, int)
True
```
It seems to me that we don't have to do long for Python 2. 
I guess I will keep
```if isinstance(start, int) and isinstance(end, int)```
?

Reply:
Yup, but I think we can still have `long` type one:

```
>>> long(1)
1L
>>> isinstance(long(1), int)
False
```

You can simply do like `isinstance(long(1), (int, long))` with 

```
if sys.version >= '3':
    long = int
```